uportal-war/src/main/java/org/jasig/portal/rendering/StructureAttributeSource.java

```diff
@@ -30,8 +32,15 @@
  * @version $Revision$
  */
 public class StructureAttributeSource extends StylesheetAttributeSource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StructureAttributeSource.class);
```

Comment:
I tend to prefer static loggers, but it's not a big deal to me.  It can go either way.  SLF4J does not recommend one over the other:  http://www.slf4j.org/faq.html#declared_static

Child classes can access the logger using the protected getLogger() method, but I would recommend that child classes define their own logger and override getLogger() to return it.

getLogger() is not so much to expose the logger to the subclasses, but rather for the superclass to use the same logger as the child so the log messages will show the subclass name.